crates/ruff_linter/src/rules/flake8_bugbear/helpers.rs

```diff
@@ -0,0 +1,28 @@
+use ruff_notebook::CellOffsets;
+use ruff_python_semantic::SemanticModel;
+use ruff_python_trivia::{SimpleTokenKind, SimpleTokenizer};
+use ruff_source_file::Locator;
+use ruff_text_size::{Ranged, TextRange};
+
+/// Return `true` if the statement containing the current expression is the last
+/// top-level expression in the cell. This assumes that the source is a Jupyter
+/// Notebook.
+pub(crate) fn at_last_top_level_expression_in_cell(
+    semantic: &SemanticModel,
+    locator: &Locator,
+    cell_offsets: Option<&CellOffsets>,
+) -> bool {
+    if !semantic.at_top_level() {
+        return false;
+    }
+    let current_statement_end = semantic.current_statement().end();
+    cell_offsets
+        .and_then(|cell_offsets| cell_offsets.containing_range(current_statement_end))
+        .is_some_and(|cell_range| {
+            !SimpleTokenizer::new(
```

Comment:
Maybe a bit easier to understand as `.all(|token| token.kind() == SimpleTokenKind::Semi || token.kind().is_trivia())`? Right now, it's a double negative with `!any` and the internal `!=`.

Reply:
Thanks! My initial implementation was using `skip_while` which clippy didn't like. This is much better.